app/models/manageiq/providers/azure/event_catcher_mixin.rb

```diff
@@ -0,0 +1,31 @@
+module ManageIQ::Providers::Azure::EventCatcherMixin
+  def parse_event_type(event)
+    event_name = event["eventName"]["value"]
+    event_type = ""
+
+    unless event["authorization"].nil? || event["authorization"]["action"].nil?
+      event_type = parse_event_action(event)
+    end
+    "#{event_type}#{event_name}"
+  end
+
+  def parse_event_action(event)
+    action = event["authorization"]["action"].split("/")
+    "#{action[1]}_#{action[2]}_"
+  end
+
+  def parse_vm_ref(event)
+    resource_id = event["resourceId"].downcase.split("/")
```

Comment:
Can you make a comment with an example of what this looks like.  The use of `resource_id[2]`, `resource_id[4]`, `resource_id[6]`, and `resource_id[7]` leave an outsider very confused about what is being joined.